README.md

```diff
@@ -10,39 +10,155 @@
 
 ## Summary
 
-A JWT issuer for the CPE devices that connect to the XMiDT cluster.
+A JWT token issuer for devices that connect to the XMiDT cluster.
 
 ## Table of Contents
 
 - [Code of Conduct](#code-of-conduct)
-- [How to Install](#how-to-install)
-- [Usage](#usage)
+- [Details](#details)
+- [Build](#build)
+- [Deploy](#deploy)
 - [Contributing](#contributing)
 
 ## Code of Conduct
 
 This project and everyone participating in it are governed by the [XMiDT Code Of Conduct](https://xmidt.io/code_of_conduct/). 
-By participating, you agree to this Code.
+By participating, you agree to this code.
 
-## How to Install
+## Details
+Themis provides a flexible strategy to issue JWT tokens to devices that need to connect to the XMiDT cluster. 
 
-### Installation
-- [Docker](https://www.docker.com/) (duh)
-  - `brew install docker`
+### Endpoints
+There are three main endpoints (directly mapped to servers `key`, `issuer` and `claims` in configuration) this service provides
 
-</br>
+- GET `/keys/{KID}`
 
-### Running
-#### Build the docker image
-```bash
-docker build -t themis:local .
+This endpoint allows fetching the public portion of the key that themis uses to sign JWT tokens. For example, [Talaria](https://github.com/xmidt-org/talaria) can use this endpoint to verify the signature of tokens which devices present when they attempt to connect to the XMiDT cloud.
+
+Configuration for this endpoint is required when the `issue` endpoint is configured and vice versa.
+
+- POST `/issue`
+
+This is the main and most compute intensive endpoint as it creates JWT tokens based on configuration. XMiDT can be configured such that it only accepts devices that have valid JWT tokens.
+
+- GET `/claims`
+
+Configuring this endpoint is required if no configuration is provided for the previous two.
+
+
+## Build
+There is a single binary for themis and its execution is fully driven by configuration.
+
+### Makefile
+
+The Makefile has the following options you may find helpful:
+* `make build`: builds the Themis binary
+* `make docker`: builds a docker image for themis
+* `make cpe-docker`: builds a docker image for themis in CPE mode
+* `make rbl-docker`: builds a docker image for themis in RBL mode 
+
```

Comment:
```suggestion
```

Reply:
Yeah, it was mostly a new standard suggestion. I reverted it back. I am sure we will revisit it as we keep adding more options (docker, rpm, helm, etc.)